#Weather app

JulianaOsemeke opened the pull request:
# weather-app
A simple weather app that extracts information from the openweathermap.org/api thereby enabling users accurately forecast weather conditions in different cities. This app was built with Javascript and Webpack with emphasis on the async/await concept.

JulianaOsemeke committed:
Set up linters

JulianaOsemeke committed:
Install webpack and dependencies

JulianaOsemeke committed:
Edit build script in package.json

JulianaOsemeke committed:
Edit package.json: Add webpack-dev-server

JulianaOsemeke committed:
Set up bare metal html

JulianaOsemeke committed:
Edit index.html

JulianaOsemeke committed:
Add styling to app

JulianaOsemeke committed:
Add icons to assets directory

JulianaOsemeke committed:
Add temp svg's to assets folder

JulianaOsemeke committed:
Add  API key to index.js

JulianaOsemeke committed:
Add promise call

JulianaOsemeke committed:
Update Assets directory

JulianaOsemeke committed:
Add event Listeners

JulianaOsemeke committed:
Create function to update weather  & Add Event listener to form

JulianaOsemeke committed:
Edit index.html: update classes

JulianaOsemeke committed:
Update weather app styling

JulianaOsemeke committed:
Clean up code

JulianaOsemeke committed:
Correct Eslint errors

JulianaOsemeke committed:
Edit: Adjust background color

JulianaOsemeke committed:
Fix ESlint errors

JulianaOsemeke committed:
Fix ESlint errors

JulianaOsemeke committed:
Edit README.md

JulianaOsemeke committed:
Edit index.js

JulianaOsemeke committed:
Edit README.md

JulianaOsemeke committed:
Edit README.md

rukundoeric commented:
Hi @JulianaOsemeke, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 

## Required Changes ♻️
  - [ ] You did put `node_modules/` in `.gitignore` but still, node_modules files are committed and pushed to your remote repo. This is because you might have added the `node_modules` to `.gitignore` after committing the files. to fix this kindly run this command. `git rm --cached -r node_modules`. (learn more [here](https://stackoverflow.com/questions/1143796/remove-a-file-from-a-git-repository-without-deleting-it-from-the-local-filesyste))
  - [ ] You overused `eslint-disable`. it's like making changes in Eslint config. they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/) and [Zendesk](https://microverse.zendesk.com/hc/en-us/articles/1500002085202-February-2nd-2021-linter-errors-as-a-reason-of-project-invalidation-)). kindy try to fix the errors instead of ignoring them.
  
  - [ ]  The app should have  a toggle control to determine if the result will be shown in Celsius or Fahrenheit (There are two valid solutions: students can make two different requests for the Celsius data and the Fahrenheit data OR make only one request and use math to convert it to the other unit) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
  - [ ] Kindly try to separate logic from DOM Manipulation. this will help you to increase the readability of your code.
  - [ ] When trying to search a city's weather, here is the error I got. kindly fix it.
  ![Screen Shot 2021-08-14 at 13 40 07](https://user-images.githubusercontent.com/45217364/129445894-08258018-2f3c-4ee3-b0de-739cd0435812.png)


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - N/A

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

rukundoeric reviewed (changes requested):
Hi @JulianaOsemeke, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 

## Required Changes ♻️
  - [x] You did put `node_modules/` in `.gitignore` but still, node_modules files are committed and pushed to your remote repo. This is because you might have added the `node_modules` to `.gitignore` after committing the files. to fix this kindly run this command. `git rm --cached -r node_modules`. (learn more [here](https://stackoverflow.com/questions/1143796/remove-a-file-from-a-git-repository-without-deleting-it-from-the-local-filesyste))
  - [x] You overused `eslint-disable`. it's like making changes in Eslint config. they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/) and [Zendesk](https://microverse.zendesk.com/hc/en-us/articles/1500002085202-February-2nd-2021-linter-errors-as-a-reason-of-project-invalidation-)). kindy try to fix the errors instead of ignoring them.
  
  - [x]  The app should have  a toggle control to determine if the result will be shown in Celsius or Fahrenheit (There are two valid solutions: students can make two different requests for the Celsius data and the Fahrenheit data OR make only one request and use math to convert it to the other unit) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
  - [x] Kindly try to separate logic from DOM Manipulation. this will help you to increase the readability of your code.
  - [x] When trying to search a city's weather, here is the error I got. kindly fix it.
  ![Screen Shot 2021-08-14 at 13 40 07](https://user-images.githubusercontent.com/45217364/129445894-08258018-2f3c-4ee3-b0de-739cd0435812.png)


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - N/A

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

JulianaOsemeke committed:
remove node modules

JulianaOsemeke committed:
Made requested changes: seperated logic from dom

JulianaOsemeke committed:
Edit index.js

JulianaOsemeke committed:
Edit README.md

JulianaOsemeke committed:
Remove node modules

JulianaOsemeke committed:
Remove node modules

JulianaOsemeke commented:
Hello. I have made the requested changes, however I have been unable to  remove the node modules with the command provided after several attempts. Thank you

afanimartin reviewed (changes requested):
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work making the requested changes. 
### Changes
- remove the node_modules folder. Learn how [here](https://github.community/t/how-to-delete-multiples-files-in-github/702)
- add a toggle button to change the temperature unit
- remove the dom-related logic in src/index on lines 15 and 16. Move that to dom.js

### Martin Afani
#### Slack: whiz25

JulianaOsemeke committed:
Add temp toggle button

JulianaOsemeke committed:
Update README.md

JulianaOsemeke committed:
Remove node modules

JulianaOsemeke commented:
Hello, I have made the following requested changes
-Removed the node modules as instructed
-Included a button that toggles between Celsius and Fahrenheit temperature units when clicked
-Removed the dom-related logic in src/index on lines 15 and 16

JulianaOsemeke committed:
Edit README.md: update live link

bellom commented on dist/index.html:
```diff
@@ -0,0 +1,60 @@
+          </div>
+          <div class="card-bottom px-5 py-4 row">
+            <div class="col text-center">
+              <p>30&deg;C</p>
```
### [Optional]
Kindly remove this hardcoded Celcius that might confuse users from the actual data.

bellom reviewed (approved):
## Hi @JulianaOsemeke  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best**
